pkgs/applications/version-management/git-and-tools/git-remote-hg/default.nix

```
@@ -1,36 +1,33 @@
-{ stdenv, lib, fetchFromGitHub, mercurial, makeWrapper
+{ stdenv, lib, fetchFromGitHub, python3Packages
 , asciidoc, xmlto, docbook_xsl, docbook_xml_dtd_45, libxslt, libxml2
 }:
 
-stdenv.mkDerivation rec {
+python3Packages.buildPythonApplication rec {
   pname = "git-remote-hg";
-  version = "1.0.1";
+  version = "unstable-2020-06-12";
 
   src = fetchFromGitHub {
     owner = "mnauw";
     repo = "git-remote-hg";
-    rev = "v${version}";
-    sha256 = "1by5ygqvq9ww990kdrjndaqsssyf2jc4n380f9pfh2avsr7871wc";
+    rev = "28ed63b707919734d230cb13bff7d231dfeee8fc";
+    sha256 = "0dw48vbnk7pp0w6fzgl29mq8fyn52pacbya2w14z9c6jfvh5sha1";
   };
 
-  buildInputs = [ mercurial.python mercurial makeWrapper
+  nativeBuildInputs = [
     asciidoc xmlto docbook_xsl docbook_xml_dtd_45 libxslt libxml2
   ];
 
-  doCheck = false;
-
-  installFlags = [ "HOME=\${out}" "install-doc" ];
+  pythonPath = with python3Packages; [ mercurial ];
```

now that it's a pythonApplication
```suggestion
  propagatedBuilInputs = with python3Packages; [ mercurial ];
```

but it doesn't really matter

---

How come it needs to be propagated? This isn’t a library, so it should only be used through the executable wrapper, which includes Mercurial anyway, right?

---

it's how python packaging works on nixpkgs https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonapplication-function, the transitive python package set is constructed off of propagated python packages.

`buildPythonApplication` will "end" this propagation to other packages which use `git-remote-hg`